### PR TITLE
Understanding min heap vs max heap in JavaScript

### PR DESCRIPTION
**Pull request by mwangiKibui**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by mwangiKibui** [comment]
This fixes #1241

**Comment by hectorkambow** [comment]
plagiarism check done

**Review by zolomohan — CHANGES_REQUESTED** [review]
@mwangiKibui  do the images used in the article need sources?

File: articles/understanding-min-heap-vs-max-heap/index.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+
+    }
+}
+```
+
+The above implementation follows the below steps:
+
+- Push the node to the end of the array.
+
+- If we more than two elements in the array, find the correct position of the value.
```
[the remark this answers is not in the document]
Comment (mwangiKibui) [reply articles/understanding-min-heap-vs-max-heap/index.md]: fixed.

File: articles/understanding-min-heap-vs-max-heap/index.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+        };
+
+        return;
+    }
+
+```
+
+The above implementation follows the below steps:
+
+- Check if the array got more than two elements. If it does not, just remove the element in the first index. If it does, continue with the below steps.
```
[the remark this answers is not in the document]
Comment (mwangiKibui) [reply articles/understanding-min-heap-vs-max-heap/index.md]: fixed.

File: articles/understanding-min-heap-vs-max-heap/index.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+        return null;
+    };
+
+    return;
+}
+```
+
+The above implementation follows the below steps:
+
+- Check if the array got more than two elements. If it does not, just remove the element in the first index. If it does, continue with the below steps.
```
[the remark this answers is not in the document]
Comment (mwangiKibui) [reply articles/understanding-min-heap-vs-max-heap/index.md]: fixed.

**Comment by mwangiKibui** [comment]
@zolomohan Have done the fixes.

File: articles/understanding-min-heap-vs-max-heap/index.md [hunk 4]
```diff
@@ -0,0 +1,520 @@
+A heap is a tree-based data structure that allows access to the minimum and maximum element in the tree in constant time. They reduce the normal run time with polynomial order. There are two types of heaps: Min-heap and Max-heap. Min-heap is used for accessing the minimum element in the heap whereas the Max-heap is used for accessing the maximum element in the heap.
```
Comment (zolomohan) [inline articles/understanding-min-heap-vs-max-heap/index.md:1]: `They reduce the normal run time with polynomial order`. Can you explain this better? What do you mean by normal runtime?

File: articles/understanding-min-heap-vs-max-heap/index.md [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+#### Retrieving the minimum element
+
+With a min-heap data structure, the minimum element is at index `1`.
+
+In the same file, under `getMin()` function, we add up the functionality:
+
+```javascript
+getMin(){
+    return heap[1];
+};
```
Comment (zolomohan) [inline articles/understanding-min-heap-vs-max-heap/index.md:126]:
```suggestion
function getMin(){
    return heap[1];
};
```

File: articles/understanding-min-heap-vs-max-heap/index.md [hunk 6]
```diff
@@ -0,0 +1,520 @@
...
+
+//testing functionality
+
+insert(10);
+insert(100);
+insert(120);
+insert(1000);
+
+console.log(getMax();
```
Comment (zolomohan) [inline articles/understanding-min-heap-vs-max-heap/index.md:376]:
```suggestion
console.log(getMax());
```

**Review by zolomohan — CHANGES_REQUESTED** [review]

File: articles/understanding-min-heap-vs-max-heap/index.md (hunk 4 above)
Comment (mwangiKibui) [reply articles/understanding-min-heap-vs-max-heap/index.md]: Solved it by explaining constant time taken.

File: articles/understanding-min-heap-vs-max-heap/index.md (hunk 5 above)
Comment (mwangiKibui) [reply articles/understanding-min-heap-vs-max-heap/index.md]: fixed

File: articles/understanding-min-heap-vs-max-heap/index.md (hunk 6 above)
Comment (mwangiKibui) [reply articles/understanding-min-heap-vs-max-heap/index.md]: fixed.

**Comment by mwangiKibui** [comment]
@zolomohan  Have done the technical fixes.

**Review by zolomohan — APPROVED** [review]

**Pull request merged**